server/src/internalClusterTest/java/org/opensearch/snapshots/SegmentReplicationSnapshotIT.java

```diff
@@ -261,4 +263,49 @@ public void testRestoreOnReplicaNode() throws Exception {
         SearchResponse resp = client().prepareSearch(RESTORED_INDEX_NAME).setQuery(QueryBuilders.matchAllQuery()).get();
         assertHitCount(resp, DOC_COUNT);
     }
+
+    public void testSnapshotRestoreOnIndexWithSegRepClusterSetting() throws Exception {
```

Comment:
Why would the behavior for snapshot/restore differ based on this setting?

Reply:
I don't think behaviour for snapshot/restore would change, but just wanted to make sure that new index restored from snapshot will use it's index settings and override the cluster setting of replication type